pkg/asset/internal/templates.go

```diff
@@ -599,6 +626,14 @@ spec:
         - name: etc-kubernetes
           mountPath: /etc/kubernetes
           readOnly: true
+      hostNetwork: true
+      tolerations:
+      - key: CriticalAddonsOnly
```

Comment:
This toleration was missing entirely from this critical pod, meaning it wouldn't get scheduled when it was dire